docs/tutorials/dataassociation/KDTree_Tutorial.py

```diff
@@ -0,0 +1,403 @@
+=================================================
+"""
+
+# %%
+# :math:`k`-d trees and TPR-trees are both data structures which can be used to index objects in multi-dimensional
+# space.
+# 
+# - A :math:`k`-d tree is a binary search tree that operates in :math:`k` dimensions where each node represents a
+#   :math:`k`-dimensional point, where :math:`k` represents any number of dimensions. A :math:`k`-d tree is useful for
+#   indexing the positions of **static** objects.
+# - A TPR-tree (a time parameterised range tree) is a binary search tree which stores information about the object's
+#   position, direction, and velocity in substructures called Velocity Bounding Rectangles (VBRs). This information can
+#   be used to compute both current and predicted positions of the object. A TPR-tree is therefore useful for indexing
+#   the positions of **moving** objects.
+# 
+# NB: a :math:`k`-d tree can also be used to index the positions of moving objects, but it will only show a snapshot of
+# positions at a given timestep. The :math:`k`-d tree must be reconstructed at each time step, but the TPR tree only
+# needs to be reconstructed if there are any changes to an object's direction and velocity.
+# 
+# Both trees can increase the efficiency of searches for information relating to a given object when compared with
+# linear searches.
+
+
+
+# %%
+# Using :math:`k`-d trees for target tracking
+# -------------------------------------------
+# So far across the other tutorials, we have seen linear Nearest Neighbour and Probabilistic Data Association search
+# algorithms applied to target tracking to identify which detections should be associated with our target track(s).
+# 
+# During training of a basic Nearest Neighbour algorithm with a given training set of detections from a sensor, the
+# algorithm simply stores the detection vectors. When the algorithm is called to identify the nearest neighbours of a
+# vector of interest (i.e. a track prediction), which we will refer to here as the **query vector**, the algorithm
+# computes the distance between each detection in the training set and the query vector. It identifies the detection
+# with the lowest distance from the query vector as the Nearest Neighbour.
+# 
+# This can be a suitable method to use for small datasets, however, as the number of detections and/or targets
+# increases, linear algorithms quickly become inefficient. A more efficient alternative to linear searches is to
+# conduct searches using a :math:`k`-d tree, which can help to quickly eliminate large sets of detections from the
+# search rather than comparing every detection to the query vector.
+
+
+# %%
+# Constructing a :math:`k`-d tree
+# -------------------------------
+# During training of an algorithm that uses a :math:`k`-d tree, the tree itself will be constructed. Trees are
+# constructed by splitting a set of detections from a sensor evenly into two groups, and then further splitting those
```

Comment:
It might bring more clarity if you remove "from a sensor" from "Trees are constructed by splitting a set of detection from a sensor evenly into two groups". The "from a sensor" component may confuse people into thinking the this method is only applicable to detections from a single _Sensor_.